src/commands/periscope/helpers/periscopehelper.ts

```diff
@@ -0,0 +1,265 @@
+import * as vscode from 'vscode';
+import * as k8s from 'vscode-kubernetes-tools-api';
+import { getSASKey, LinkDuration } from '../../utils/azurestorage';
+import { parseResource } from '../../../azure-api-utils';
+import * as ast from 'azure-arm-storage';
+import { PeriscopeStorage, PeriscopeHTMLInterface } from '../models/storage';
+import * as amon from 'azure-arm-monitor';
+import * as htmlhandlers from "handlebars";
+import * as path from 'path';
+import * as fs from 'fs';
+import AksClusterTreeItem from '../../../tree/aksClusterTreeItem';
+const tmp = require('tmp');
+
+const {
+    BlobServiceClient,
+    StorageSharedKeyCredential
+} = require("@azure/storage-blob");
+
+export async function getClusterDiagnosticSettings(
+    cluster: AksClusterTreeItem
+): Promise<amon.MonitorManagementModels.DiagnosticSettingsCategoryResourceCollection | undefined> {
+    try {
+        // Get daignostic setting via diagnostic monitor
+        const diagnosticMonitor = new amon.MonitorManagementClient(cluster.root.credentials, cluster.root.subscriptionId);
+        const diagnosticSettings = await diagnosticMonitor.diagnosticSettingsOperations.list(cluster.id!);
+
+        return diagnosticSettings;
+    } catch (e) {
+        vscode.window.showErrorMessage(`Error fetching cluster diagnostic monitor: ${e}`);
+        return undefined;
+    }
+}
+
+export async function chooseStorageAccount(
+    diagnosticSettings: amon.MonitorManagementModels.DiagnosticSettingsResourceCollection | undefined,
```

Comment:
Do you actually need to allow `undefined` as input here?  Consider making the argument 

```suggestion
    diagnosticSettings: amon.MonitorManagementModels.DiagnosticSettingsResourceCollection,
```

and bailing early at the call site.